### ISSUE TITLE
Reproducing results from DMS datasets

### ISSUE THREAD

**Issue by mat10d** [brianhie/efficient-evolution#25]
I was able to run the first part of `bash bin/dms.sh` locally (that calls `dms.py`) but I'm struggling to understand how I can reproduce the manuscript results. It seems that for all of these datasets, (except infA, for which 0.5 was used), alpha was set to 1. However, how can I know what value of k is being used, and what is the best way to specify k when calling dms.py/reconstruct_multi_models? Thank you in advance!

**Comment by brianhie** [comment]
Hi @mat10d, the k cutoff values are the fourth field in the list here: https://github.com/brianhie/efficient-evolution/blob/36d06a0cc84d5493655dbf12862bec4553b92d5e/bin/plot_dms_esm_scatter.py#L43. We used either k = 1 or 2.

**Comment by mat10d** [comment]
Hey @brianhie -- are there any differences from when you run the model using the dms.sh vs. just getting the raw sequences and running through recommend.py? I know the alpha setting is smaller for one dataset, but when I run the the attached bash file relying on recommend.py, I get the following recommendations (txt file). This doesn't correspond to the number of mutants that are presented in the final paper or that are outputted by dms.sh. I wanted to generate those exactly so I could also run recommend.py with similar settings for further dms experiments (cov2_S, zikv_E, cas12f for example). I'd assume that this is down to some parameter changes but I haven't been able to identify them. Thanks in advance! The notebook also attached shows how using the k-cutoffs for both outputs generate different number of suggested mutants.

[efficient-evolution.zip](https://github.com/user-attachments/files/15527168/efficient-evolution.zip)

**Comment by brianhie** [comment]
Hi @mat10d -- I took a quick look and I'm not sure. Are you getting different results when you do dms.sh and recommend.py with the same alpha parameters? They are fundamentally calling the same function (reconstruct_multi_models) so I would take a look at the inputs and outputs to that. I think recommend.py should be considered the "reference" implementation if you want to use the pipeline.

**Comment by mat10d** [comment]
Hey Brian, I took a better look, and what I realized is that when I use recommend.py, I am getting AA positions where the model is recommending more than 1 substitution, for example, for mapk1, it recommends 3 substitutions at position 11: P11G, P11K, P11R. From what I can tell, the output of dms.sh is only one substitutions out of the 3. Are you constraining for just the top substitution out of the available ones?

Regardless, I can see that the results beyond this are equivalent, so maybe I can work with what I am getting out of recommend.py anyway, if you consider it to be the "reference" implementation!
